offline_optimization_script.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+"""Offline optimization of parameter combinations."""
+
+import argparse
+import itertools
+import logging
+from pathlib import Path
+from typing import Literal
+
+import numpy as np
+
+from ptxboa import DEFAULT_CACHE_DIR, DEFAULT_DATA_DIR
+from ptxboa.api import PtxboaAPI
+
+
+def product_dict(**kwargs):
+    """Yield the cartesian product of a dictionary of lists.
+
+    https://stackoverflow.com/a/5228294
+    """
+    keys = kwargs.keys()
+    for instance in itertools.product(*kwargs.values()):
+        yield dict(zip(keys, instance))
+
+
+def main(
+    cache_dir: Path = DEFAULT_CACHE_DIR,
+    loglevel: Literal["debug", "info", "warning", "error"] = "info",
+):
+    fmt = "[%(asctime)s %(levelname)7s] %(message)s"
+    datefmt = "%Y-%m-%d %H:%M:%S"
+    logging.basicConfig(level=loglevel.upper(), format=fmt, datefmt=datefmt)
+    api = PtxboaAPI(data_dir=DEFAULT_DATA_DIR, cache_dir=cache_dir)
+
+    # these are the parameter dimensions that are relevant for the optimization
+    param_arrays = {
+        "region": api.get_dimension("region")["region_name"].tolist(),
+        "scenario": api.get_dimension("scenario").index.tolist(),
+        "secproc_water": api.get_dimension("secproc_water").index.tolist(),
+        "country": api.get_dimension("country").index.tolist(),
+        "res_gen": api.get_dimension("res_gen").index.tolist(),
+        "secproc_co2": api.get_dimension("secproc_co2").index.tolist(),
+        "chain": api.get_dimension("chain").index.tolist(),

```

Comment:
- ``country`` is not relevant
- `´secproc_water`` and ``secproc_co2`` are not relevant at the moment, because the distinction between the two possible settings is not yet implemented in the optimization module. They will become relevant later, though #330